02_cancer_type_classification/run_cancer_type_classification.py

```diff
@@ -49,6 +54,12 @@ def process_args():
                         'cancer types in TCGA if none are provided')
     p.add_argument('--log_file', default=None,
                    help='name of file to log skipped cancer types to')
+    p.add_argument('--mad_preselect', type=int, default=None,
+                   help='if included, pre-select this many features by MAD, '
+                        'before applying primary feature selection method')
+    p.add_argument('--num_features', type=int, default=cfg.num_features_raw,
```

Comment:
What's the difference between mad_preselect and num_features? Their description reads the same to me.

Reply:
`mad_preselect` happens _before_ `feature_selection`/`num_features`. It's mainly a way to speed things up a bit for our early experiments here - since calculating the f-statistic for each of 16,000 features in each cancer type can be slow, sometimes it makes sense to start with (for example) the top 5,000 most variable features, then do the univariate correlation calculations only on those features. I'll try to make this clearer in the help strings.